src/pypromice/process/L2toL3.py

```diff
@@ -315,72 +388,82 @@ def calcHumid(T_0, T_100, T_h, es_0, es_100, eps, p_h, RH_cor_h):
+    done piecewise so that each period between station relocations are done 
+    separately (no smoothing of the jump due to relocation). Piecewise linear 
+    regression is then used to smooth the available observations. Then this 
+    smoothed curve is interpolated linearly over internal gaps. Eventually, this 
+    interpolated curve is extrapolated linearly for timestamps before the first 
+    valid measurement and after the last valid measurement.
     
     Parameters
     ----------
-    SHF : xarray.DataArray
-        Sensible heat flux
-    LHF : xarray.DataArray
-        Latent heat flux
-    T : xarray.DataArray
-        Air temperature
-    Tsurf : xarray.DataArray
-        Surface temperature
-    p : xarray.DataArray
-        Air pressure
-    WS : xarray.DataArray
-        Wind speed
-    RH_cor : xarray.DataArray
-        Relative humidity corrected
-    z_boom : xarray.DataArray
-        Boom height
-    
+    df_in : pandas.Series
+        Series of observed latitude, longitude or elevation with datetime index.
+    breaks: list
+        List of timestamps of station relocation. First and last item should be
+        None so that they can be used in slice(breaks[i], breaks[i+1])
+        
     Returns
     -------
-    SHF : xarray.DataArray
-        Sensible heat flux corrected
-    LHF : xarray.DataArray
-        Latent heat flux corrected
+    np.ndarray
+        Smoothed and interpolated values corresponding to the input series.
     '''
-    HF_nan = np.isnan(p) | np.isnan(T) | np.isnan(Tsurf) \
-        | np.isnan(RH_cor) | np.isnan(WS) | np.isnan(z_boom)
-    SHF[HF_nan] = np.nan
-    LHF[HF_nan] = np.nan 
-    return SHF, LHF
-      
-def cleanSpHumid(q_h, T, Tsurf, p, RH_cor):
-    '''Find invalid specific humidity data values and replace with NaNs, 
-    based on air temperature, surface temperature, air pressure, 
-    and corrected relative humidity
+    df_all = pd.Series(dtype=float)  # Initialize an empty Series to gather all smoothed pieces
+    breaks = [None] + breaks + [None]
+    for i in range(len(breaks) - 1):
+        df = df_in.loc[slice(breaks[i], breaks[i+1])].copy()
+               
+        # Drop NaN values and calculate the number of segments based on valid data
+        df_valid = df.dropna()
+        if df_valid.shape[0] > 2:        
+            # Fit linear regression model to the valid data range
+            x = pd.to_numeric(df_valid.index).values.reshape(-1, 1)
+            y = df_valid.values.reshape(-1, 1)
+            
+            model = LinearRegression()
+            model.fit(x, y)
+            
+            # Predict using the model for the entire segment range
+            x_pred = pd.to_numeric(df.index).values.reshape(-1, 1)
+            
+            y_pred = model.predict(x_pred)
+            df =  pd.Series(y_pred.flatten(), index=df.index)
+
+        
+        # Update df_all with predicted values for the current segment
+        df_all = pd.concat([df_all, df])
```

Comment:
For inspiration:

Consider creating a list of output inferred series instead of updating df_all repeatedly.
```python
_infered_series = []
for break in ..:
    # do the model fitting and inference
    _infered_series.append(df)    
df_all = pd.concat(_infered_series)
```